aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -988,3 +1011,107 @@ impl AsMut<AptosVMImpl> for AptosVM {
         &mut self.0
     }
 }
+
+impl AptosSimulationVM {
+    fn validate_simulated_transaction<S: MoveResolverExt>(
+        &self,
+        session: &mut SessionExt<S>,
+        transaction: &SignedTransaction,
+        txn_data: &TransactionMetadata,
+        allow_too_new: bool,
+        log_context: &AdapterLogSchema,
+    ) -> Result<(), VMStatus> {
+        self.0.check_transaction_format(transaction)?;
+
+        let prologue_status =
+            self.0
+                .run_prologue_with_payload(session, transaction.payload(), txn_data, log_context);
+        match prologue_status {
+            Err(err)
+                if !allow_too_new || err.status_code() != StatusCode::SEQUENCE_NUMBER_TOO_NEW =>
+            {
+                Err(err)
+            }
+            _ => Ok(()),
+        }
+    }
+
+    /*
+    Executes a SignedTransaction without performing signature verification
+     */
+    fn simulate_signed_transaction<S: MoveResolverExt>(
+        &self,
+        storage: &S,
+        txn: &SignedTransaction,
+        log_context: &AdapterLogSchema,
+    ) -> (VMStatus, TransactionOutput) {
+        macro_rules! unwrap_or_discard {
+            ($res: expr) => {
+                match $res {
+                    Ok(s) => s,
+                    Err(e) => return discard_error_vm_status(e),
+                }
+            };
+        }
```

Comment:
probably fine to just make this inline rather than a macro as that will make it more readable. Plus you only use this once. You'll notice we don't use a lot of macros in the codebase.